x/mongo/driver/connstring/connstring.go

```diff
@@ -762,6 +771,12 @@ func (p *parser) addOption(pair string) error {
 		}
 		p.SocketTimeout = time.Duration(n) * time.Millisecond
 		p.SocketTimeoutSet = true
+	case "srvservicename":
+		// srvServiceName can only be set on URIs with the "mongodb+srv" scheme
+		if p.Scheme != SchemeMongoDBSRV {
+			return fmt.Errorf("cannot specify srvServiceName on non-SRV URI")
+		}
```

Comment:
Should there be a check to ensure that `srvServiceName` is >= 1 character and <= 62 characters?

Reply:
Good catch! Added.